akka-persistence-query/src/main/scala/akka/persistence/query/javadsl/Offset.scala

```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright (C) 2015-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.persistence.query.javadsl
+
+import java.util.UUID
+
+trait Offset {
+  def hashCode: Int
+  def asScala: akka.persistence.query.scaladsl.Offset
```

Comment:
I don't think we need different classes for scaladsl and javadsl. We can place it in `akka.persistence.query` package.